### DISCUSSION TITLE
Getting MIDI data/music from Piano to corresponding LED lights and PC for recording.

### DISCUSSION THREAD

**Discussion by StarDreamIX** [onlaj/Piano-LED-Visualizer#494] [Q&A] [answered]
Hello all,
I'm having some trouble in getting my piano to transmit MIDI data to my computer:

I have a keyboard of which I wanted to do 2 things at the same time:
1) Work with my individually addressable LED lights (already working)
2) transmit MIDI data to computer for music (not working)

So,

I have a piano model (DONNER DEP-10) which I want to connect a USB midi cable to my computer to record music, but I have already used the port on connecting a MIDI cable to a raspberry pi 3B for controlling individually addressable LED lights such that when I press a key, the corresponding light flashes, which works fine!! - (using your OS for my piano/lights.)

![light](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/a3ad9f1a-8e9c-4dab-ac81-593ef1ace1f6)

The problem is that I now there can't connect any cables to my computer to allow recording of MIDI data - as the usb midi port on my piano is now taken:

![taken](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/dacc3de1-9777-4f44-b92c-3296c6922933)

![midi taken](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/c29f546a-d5d6-4346-964c-b2060575f8c8)

![Screenshot 2023-12-23 161223](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/42ab6382-0774-41f1-bac9-df3278402796)
___
What I was thinking:

I've been looking in the manual for my piano (which i'll leave attached) (and in image above) - the PEDAL socket suspiciously looks like an old MIDI cable socket where an old MIDI cable would go. But I don't know if it's MIDI IN, MIDI OUT or MIDI THRU, or if it even transmits MIDI data at all.

Also the usb socket in my piano as far as I know only allows the playing of .mp3 songs on a USB plugged into it. - It doesn't record any midi data :(
____

how do I get the midi data to register and play on my computer while simultaneously lighting my lights correctly? other forums directed me to buying one of these MIDI to USB interface cables:

![Screenshot 2023-12-23 165423](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/35edd610-51bc-4aaa-8873-49b7ddecd17d)

but there is only one socket on my piano where it could fit and that's either midi in, out, thru, or doesnt work at all.

Here is the raspberry pi 3b which has a custom operating system allowing my lights to work:

![pi](https://github.com/onlaj/Piano-LED-Visualizer/assets/137229268/6e26f52b-1cbb-4323-9f3c-6836ebeb68f3)

I looked at various other links if your interested [PiMiDi: a Raspberry Pi Midi Box, or How I Learned to Stop Worrying and Love MIDI : 5 Steps - Instructables](https://www.instructables.com/PiMiDi-A-Raspberry-Pi-Midi-Box-or-How-I-Learned-to/) 
But I would have to get another raspberry pi for it to work.(i've spent enough money already!) and the OS on my raspberry pi which gets my lights working, I have no Idea if putting the midi board on the raspberry pi would help? [MIDI Board for Raspberry Pi | The Pi Hut](https://thepihut.com/products/midi-board-for-raspberry-pi) which column would I put it on and if my piano (pedal) output doesnt actually work, then the board isnt useful.

Any methods on what I should do to get both things working?

Piano: Donner DEP-10 (I attached my manual of my piano below) (look on pages 3 and 4 for socket information)
Raspberry pi model: 3B
Lights: LED Strip WS2812B 5050 RGB (working)
OS that allows my lights to work (flashed on sd card in raspberry pi): [onlaj/Piano-LED-Visualizer: Piano LED Visualizer: Connect an LED strip to your Raspberry Pi and create an immersive visual experience for your piano playing (github.com)](https://github.com/onlaj/Piano-LED-Visualizer)
Music software im using: mixcraft 8 pro studio

Thanks so much! And have an amazing Christmas all!
[dep10.pdf](https://github.com/onlaj/Piano-LED-Visualizer/files/13759745/dep10.pdf)

**Comment by onlaj** [comment, marked as answer]
Hello
You can find an answer here: https://github.com/onlaj/Piano-LED-Visualizer/blob/master/Docs/features.md#we-need-to-make-a-connection-between-your-pcmacandroid-there-are-at-least-3-ways-of-doing-that
Merry Christmas!

**Reply by StarDreamIX** [reply]
thank you very much! - so If I buy the Sevilla's Soft MIDI USB-USB device via https://compasflamenco.com/midi-usbusb-p-4.html?cPath=3
and connect it from my pi to my pc, with your os running on the pi, it should transmit the MIDI data? - just double checking before ordering, thanks!

**Reply by onlaj** [reply]
Yes, that's what I use.
So far this is the most reliable and easiest option, although a little expensive.
